activation/interface.go

```diff
@@ -53,9 +53,10 @@ type nipostBuilder interface {
 		ctx context.Context,
 		sig *signing.EdSigner,
 		publish types.EpochID,
-		challenge types.Hash32,
+		challengeHash types.Hash32,
+		postChallenge *types.NIPostChallenge,
```

Comment:
NIT: technically its a poet challenge. The hash of it is sent to the poet service. It collects the data that needs to be proven that was selected and known before registering to the poet.

The challenge for PoST is the hash of the poet proof.

```suggestion
		nipostChallenge *types.NIPostChallenge,
```

Reply:
Not sure if I fully understand, if it is a poet challenge then why are you suggesting to rename it to `nipostChallenge`? and is there a semantic difference between a `postChallenge` and a `nipostChallenge`?

Reply:
The hash of the nipost challenge is the one submitted to the poet (that's why I said "poet challenge", a "challenge for the poet"). 

> is there a semantic difference between a postChallenge and a nipostChallenge?

The naming is confusing, we have too many "challenges". The simplified flow is:
1. build nipost challenge
2. submit hash of nipost challenge to the poet
3. wait 2w, obtain the poet proof
4. use the root of the poet proof as the challenge to generate the PoST ("post challenge"): https://github.com/spacemeshos/go-spacemesh/blob/7b0999303a2f4ccf4267c98f30489666909b1c62/activation/nipost.go#L290